packages/react/src/prefabs/Chat.tsx

```diff
@@ -4,25 +4,30 @@ import * as React from 'react';
 import { useMaybeLayoutContext, useRoomContext } from '../context';
 import { useObservableState } from '../hooks/internal/useObservableState';
 import { cloneSingleChild } from '../utils';
-import type { MessageFormatter } from '../components/ChatEntry';
+import type { MessageDecoder, MessageEncoder, MessageFormatter } from '../components/ChatEntry';
 import { ChatEntry } from '../components/ChatEntry';
 
 export type { ChatMessage, ReceivedChatMessage };
 
 /** @public */
 export interface ChatProps extends React.HTMLAttributes<HTMLDivElement> {
   messageFormatter?: MessageFormatter;
+  messageEncoder?: MessageEncoder;
+  messageDecoder?: MessageDecoder;
 }
 
 /** @public */
-export function useChat() {
+export function useChat(options: {
```

Comment:
the options object needs to be optional

Reply:
I think fallbacking to empty object is better, so I can still do 
```const { messageDecoder, messageEncoder } = options;```

Reply:
you can do the fallback in `core`, but in the react package that unfortunately causes unneeded re-evaluations because (for some reason) default parameters in react like objects and arrays are always considered as "new" objects. 

you'll also want to include `options` in the useEffect dependency array

Reply:
Hi @lukasIO I've updated all comments can you recheck the PR?